app/assets/javascripts/3dbio_viewer/src/webapp/components/annotations-tool/AnnotationsTool.tsx

```diff
@@ -1,71 +1,87 @@
-import React, { useState, useRef } from "react";
+import React, { useCallback, useState, useRef } from "react";
 import _ from "lodash";
 import { Dialog, DialogContent, DialogTitle, IconButton, Switch } from "@material-ui/core";
 import { Close } from "@material-ui/icons";
 import i18n from "../../utils/i18n";
+import { useBooleanState } from "../../hooks/use-boolean";
 import { Dropzone, DropzoneRef } from "../dropzone/Dropzone";
 import { ProtvistaAction } from "../protvista/Protvista.helpers";
 import "./AnnotationsTool.css";
 
-export interface ModelUploadProps {
-    title: string;
+export interface AnnotationsToolProps {
     onClose(): void;
     action: ProtvistaAction;
 }
 
-export const AnnotationsTool: React.FC<ModelUploadProps> = React.memo(props => {
-    const { title, onClose, action } = props;
+interface AnnotationForm {
+    trackName: string;
+    type: string;
+    description: string;
+    color: string;
+    index: string;
+    startingValue: number;
+    endingValue: number;
+}
+
+export const AnnotationsTool: React.FC<AnnotationsToolProps> = React.memo(props => {
+    const { onClose, action } = props;
 
     const annotationFileRef = useRef<DropzoneRef>(null);
     const [error, setError] = useState<string>();
-    const [isManual, setIsManual] = useState<boolean>(false);
-    const [type, setType] = useState<string>("Region");
-    const [description, setDescription] = useState<string>("");
-    const [color, setColor] = useState<string>("");
-    const [startingValue, setStartingValue] = useState<number>();
-    const [endingValue, setEndingValue] = useState<number>();
+    const [isManual, { toggle: toggleIsManual }] = useBooleanState(false);
+    const [annotationForm, setAnnotationForm] = useState<AnnotationForm>({
+        trackName: action.trackId,
+        type: "",
+        description: "",
+        color: "",
+        index: "sequence",
+        startingValue: 0,
+        endingValue: 0,
+    });
 
-    const addManualAnnotation = () => {
+    const addManualAnnotation = useCallback(() => {
         setError("");
-        if (!startingValue) {
+        if (!annotationForm.startingValue) {
             setError("Missing starting value: please fill in a starting value.");
         }
-        if (!endingValue) {
-            setEndingValue(startingValue);
+        if (!annotationForm.endingValue) {
```

Comment:
I guess that 0 is not a valid value, otherwise we'd check against 'undefined`, as 0 is falsy.